src/lib.rs

```diff
@@ -807,7 +828,7 @@ impl<EF: ExtraTokenFields, TT: TokenType, TE: ErrorResponseType> Client<EF, TT,
     pub fn exchange_code(
         &self,
         code: AuthorizationCode
-    ) -> Result<TokenResponse<EF, TT>, RequestTokenError<TE>> {
+    ) -> impl Future<Item = TokenResponse<EF, TT>, Error = RequestTokenError<TE>> + Send + 'static {
```

Comment:
tokio requires [`send + 'static`](https://tokio.rs/blog/2018-03-tokio-runtime/) because it adopts the [work-stealing algorithm](https://tokio-rs.github.io/tokio/tokio/runtime/fn.run.html).